File: core/src/main/java/org/tgoe/core/member/validation/Validator.java
package org.tgoe.core.member.validation;

import java.util.ArrayList;
import java.util.List;

public abstract class Validator {
	protected List<ValidationMessage> messages = new ArrayList<ValidationMessage>();
	protected abstract String getValidationName();
	
	public List<ValidationMessage> getMessages() {
		return messages;
	}
	
	protected void addMessage(String objectName, ValidationSeverity severity, String messageText) {
		messages.add(new ValidationMessage(severity, objectName, getValidationName(), messageText));
	}	
	
}

File: src/TgoeSrv/Member/Validator/Validator.php
<?php
declare(strict_types = 1);
namespace TgoeSrv\Member\Validator;

use TgoeSrv\Member\Enums\ValidationSeverity;

abstract class Validator
{

    protected array $validatorMessages = array();

    protected abstract function getValidatorName(): string;

    protected function addMessage(ValidationSeverity $severity, object $targetObject, string $message): void
    {
        $this->validatorMessages[] = new ValidationMessage($severity, $this->getValidatorName(), $targetObject, $message);
    }

    public function getMessages(): array
    {
        return $this->validatorMessages;
    }
}
?>